dec_factory.py

```diff
@@ -0,0 +1,47 @@
+# Omer Gertler
+# decorator factory for checking parameter's type
+
+from functools import wraps
+
+
+def type_check(correct_type):
+    """ Check type of an argument"""
+    def dec(fn):
+        @wraps(fn)
+        def inner(arg):
+            """  A wrapper function. """
+            if isinstance(arg, correct_type):
+                res = fn(arg)
+                print(res)
+                print(f'Check-type result: {True} (Except: {correct_type}, got: {correct_type}\n')
+            else:
+                raise CustomError(correct_type, type(arg))
+        return inner
+    return dec
```

Comment:
'fn', 'res' 'dec' arent explanatory names
 Variable names - https://workat.tech/machine-coding/tutorial/writing-meaningful-variable-names-clean-code-za4m83tiesy0 - variable names, abbreviation reference is missing